gen/generate_test.go

```diff
@@ -79,6 +79,62 @@ func TestGenerateWithRelativePaths(t *testing.T) {
 	}
 }
 
+func TestGenerateWithHyphenPaths(t *testing.T) {
+	outputDir, err := ioutil.TempDir("", "thriftrw-generate-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(outputDir)
+
+	thriftRoot, err := os.Getwd()
+	require.NoError(t, err)
+
+	tests := []struct {
+		filepath      string
+		generateError string
+		compileError  string
+	}{
+		{
+			filepath:     "internal/tests/thrift/nestedfiles_error/include_hyphen_files.thrift",
+			compileError: "including hyphenated thrift file is not allowed",
+		},
+		{
+			filepath: "internal/tests/thrift/abc-defs.thrift",
+		},
+		{
+			filepath:      "internal/tests/thrift/nestedfiles_error/abc-defs-nested.thrift",
+			generateError: "is colliding with existing file",
+		},
+		{
+			filepath: "internal/tests/thrift/nestedfiles_error/abc_defs_nested.thrift",
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
style nit: generally seen the following at Uber: 
```suggestion
	for _, tt := range tests {
```
https://github.com/uber-go/guide/blob/master/style.md#test-tables

Reply:
usages need to renamed too. hence would do in a commit